src/opentime/rationalTime.cpp

```diff
@@ -94,6 +94,103 @@ is_dropframe_rate(double rate)
     return std::find(b, e, rate) != e;
 }
 
+static bool
+parseFloat(char const* pCurr, char const* pEnd, bool allow_negative, double* result)
+{
+    if (pCurr >= pEnd || !pCurr)
+    {
+        *result = 0.0;
+        return false;
+    }
+
+    double ret = 0.0;
+    double sign = 1.0;
+
+    if (*pCurr == '+')
+    {
+        ++pCurr;
+    }
+    else if (*pCurr == '-') 
+    {
+        if (!allow_negative) 
+        {
+            *result = 0.0;
+            return false;
+        }
+        sign = -1.0;
+        ++pCurr;
+    }
+
+    // get integer part
+    //
+    // Note that uint64_t is used because overflow is well defined for
+    // unsigned integers, but it is undefined behavior for signed integers,
+    // and floating point values are couched in the specification with
+    // the caveat that an implementation may be IEEE-754 compliant, or only
+    // partially compliant.
+    //
+    uint64_t uintPart = 0;
+    while (pCurr < pEnd) 
+    {
+        char c = *pCurr;
+        if (c < '0' || c > '9')
+        {
+            break;
+        }
+        uint64_t accumulated = uintPart * 10 + c - '0';
+        if (accumulated < uintPart)
+        {
+            // if there are too many digits, resulting in an overflow, fail
+            *result = 0.0;
+            return false;
+        }
+        uintPart = accumulated;
+        ++pCurr;
+    }
+
+    ret = (double) uintPart;
```

Comment:
You might want to consider a static_cast here instead of a C cast -- static_casts are verified at compile team and C casts at runtime (and can thus fail).

Reply:
sounds good, done